### PR TITLE
Fix upload of new packages

### LINKED ISSUE #7022: Cannot upload mods in Package Manager (403 Error)
#### Description
After installing latest Github build 697d1f3 on my test forum, no one mod package cannot be uploaded in Package Manager.
I get 'Error 403' on the white screen. Reinstalling RC4 resolve this problem.

### Steps to reproduce
1. Install latest build 697d1f3
2. Open Package Manager and try to upload some mod package

### Environment (complete as necessary)
- Version/Git revision: 697d1f3
- Database Type: MariaDB
- Database Version: 10.3.31
- PHP Version: 8.0.9

### Additional information/references

### PR DESCRIPTION
**Pull request by BrickOzp**
Also removed logging of errors.

Fixes #7022

Signed-off-by: Oscar Rydhé <oscar.rydhe@gmail.com>

### CODE REVIEW

File: Sources/Subs-Package.php [hunk 1]
```diff
@@ -3016,13 +3016,10 @@ function package_crypt($pass)
  */
 function package_unique_filename($dir, $filename, $ext)
 {
-	if (file_exists($dir . '/' . $filename . '.' . $ext))
```
Comment (BrickOzp) [inline Sources/Subs-Package.php]: At the time this function is called, the file was not moved to the destination yet. So it does not exists and the input filename will always be used.

File: Sources/PackageGet.php [hunk 2]
```diff
@@ -673,18 +673,19 @@ function PackageUpload()
...
-		fatal_lang_error('package_upload_error_exists');
+		fatal_lang_error('package_upload_error_exists', false);
 
 	// Now move the file.
 	move_uploaded_file($_FILES['package']['tmp_name'], $destination);
 	smf_chmod($destination, 0777);
 
 	// If we got this far that should mean it's available.
-	$context['package'] = getPackageInfo($packageName);
+	$context['package'] = getPackageInfo($packageFileName);
```
Comment (BrickOzp) [inline Sources/PackageGet.php:688]: Need to pass the filename with extension here

File: Sources/PackageGet.php [hunk 3]
```diff
@@ -700,7 +701,7 @@ function PackageUpload()
 	{
 		while ($package = readdir($dir))
 		{
-			if ($package == '.' || $package == '..' || $package == 'temp' || $package == $packageName || (!(is_dir($packagesdir . '/' . $package) && file_exists($packagesdir . '/' . $package . '/package-info.xml')) && substr(strtolower($package), -7) != '.tar.gz' && substr(strtolower($package), -4) != '.tgz' && substr(strtolower($package), -4) != '.zip'))
+			if ($package == '.' || $package == '..' || $package == 'temp' || $package == $packageFileName || (!(is_dir($packagesdir . '/' . $package) && file_exists($packagesdir . '/' . $package . '/package-info.xml')) && substr(strtolower($package), -7) != '.tar.gz' && substr(strtolower($package), -4) != '.tgz' && substr(strtolower($package), -4) != '.zip'))
```
Comment (BrickOzp) [inline Sources/PackageGet.php:704]: Need to check with extension here as well

**Comment by BrickOzp** [comment]
@live627 can you check this, since it modifies your commit

File: Sources/Subs-Package.php (hunk 1 above)
Comment (live627) [reply Sources/Subs-Package.php]: The extra number should only be appended if the input file does not exist, similar  to how the package backup does it.
https://github.com/SimpleMachines/SMF2.1/blob/4bbda49cf841a5008539a72b75015dce610e31d4/Sources/Subs-Package.php#L3099-L3105

Comment (BrickOzp) [reply Sources/Subs-Package.php]: Makes sense. I have put it back

**Review by live627 — APPROVED** [review]

**Pull request merged**